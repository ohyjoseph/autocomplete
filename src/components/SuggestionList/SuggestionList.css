.root {
  list-style: none;
  padding-left: 0;
  width: 200px;
  border: 1px solid #ccc;
  box-shadow: 1px 1px 3px #ccc;
  color: #333;
  font-size: 1em;
  font-family: monospace;
}

.root li {
  padding: 3px;
  border-bottom: 1px solid #ccc;
}
