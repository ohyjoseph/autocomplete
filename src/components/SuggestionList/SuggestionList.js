import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import s from './SuggestionList.css';
import { Suggestion } from '../../suggestion/core';

class SuggestionList extends React.Component {
  static propTypes = {
    suggestions: PropTypes.arrayOf(PropTypes.instanceOf(Suggestion)).isRequired,
  };

  static defaultProps = {
    onClick: null,
  };
  render() {
    const { suggestions } = this.props;
    return (
      <ul className={s.root}>
        {suggestions.map(suggestion =>
          <li key={suggestion.snippet}>
            {suggestion.snippet}
          </li>,
        )}
      </ul>
    );
  }
}

export default withStyles(s)(SuggestionList);
