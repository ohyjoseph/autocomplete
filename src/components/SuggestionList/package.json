{
  "name": "SuggestionList",
  "version": "0.0.0",
  "private": true,
  "main": "./SuggestionList.js"
}
