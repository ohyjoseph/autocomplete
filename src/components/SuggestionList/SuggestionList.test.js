/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import SuggestionList from './SuggestionList';

describe('SuggestionList', () => {
  test('renders empty correctly', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {} }}>
          <SuggestionList suggestions={[]} />
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
