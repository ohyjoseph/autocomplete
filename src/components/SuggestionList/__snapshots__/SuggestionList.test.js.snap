// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionList renders empty correctly 1`] = `
<ul
  className="root"
/>
`;
